include/keisan/matrix/matrix.hpp

```diff
@@ -27,9 +27,19 @@ namespace keisan
 {
 
 // Forward declaration
+struct Point2;
+
+struct Point3;
+
 template<size_t N>
 class Vector;
 
+template<typename T>
+class Angle;
+
+template<typename T>
+class Euler;
```

Comment:
because `Point2`, `Point3`, `Vector<N>`, `Angle<T>` and `Euler<T>` doesn't depend on `matrix.hpp`, i guess it's safe to just include all of those headers here, instead of forward declared it.

Reply:
Oh yeah, I've just realized that the problem is in the `point` header files. Those files include `matrix` header file.